.circleci/config.yml

```diff
@@ -0,0 +1,79 @@
+version: 2
+jobs:
+  build:
+    parallelism: 1
+    docker:
+      - image: circleci/elixir:1.4-node-browsers
+        environment:
+          MIX_ENV: test
+          CLIENT_ID: fake
+          CLIENT_SECRET_ID: fake
+          OUTBOUND_EMAIL_DOMAIN: http://localhost
+          INBOUND_EMAIL_DOMAIN: http://localhost
+          SLACK_WEBHOOK_URL: https://fakeslack/webhook
+      - image: circleci/postgres:9.5
+        environment:
+          POSTGRES_USER: postgres
+          POSTGRES_DB: constable_api_test
+          POSTGRES_PASSWORD:
+
+    working_directory: ~/app
+
+    steps:
+      - checkout
+
+      - run: mix local.hex --force
+      - run: mix local.rebar --force
+
+      - restore_cache:
+          keys:
+            - v1-mix-cache-{{ .Branch }}-{{ checksum "mix.lock" }}
+            - v1-mix-cache-{{ .Branch }}
+            - v1-mix-cache
+      - restore_cache:
+          keys:
+            - v1-build-cache-{{ .Branch }}
+            - v1-build-cache
+      - run: mix do deps.get, compile --warnings-as-errors
+      - save_cache:
+          key: v1-mix-cache-{{ .Branch }}-{{ checksum "mix.lock" }}
+          paths: "deps"
+      - save_cache:
+          key: v1-mix-cache-{{ .Branch }}
+          paths: "deps"
+      - save_cache:
+          key: v1-mix-cache
+          paths: "deps"
+      - save_cache:
+          key: v1-build-cache-{{ .Branch }}
+          paths: "_build"
+      - save_cache:
+          key: v1-build-cache
+          paths: "_build"
+
+      - restore_cache:
+          keys:
+            - v1-npm-cache-{{ .Branch }}-{{ checksum "assets/package-lock.json" }}
+            - v1-npm-cache-{{ .Branch }}
+            - v1-npm-cache
+      - run:
+          working_directory: assets
+          command: npm install && node_modules/.bin/brunch build
```

Comment:
I don't know how important it is to enforce the `node` or `npm` versions across environments, but here you're using whatever circleci chose to install in their image, which doesn't look configurable, and _may_ not be a version of `npm` that looks at the `package-lock.json`. I figure if it's any version of `npm` that uses the `package-lock.json`, then it's safe; but if they have an older version that doesn't look at `package-lock.json`, then you might experience inconsistent browser-based tests because it's installing different versions that what's locked.

I took the extra steps in another project to install `asdf` during this setup process and only use it for `nodejs`, but it stinks to have to do that.

Reply:
Yeah, I was really trying to avoid having to install. I was assuming/hoping it would just use `package-lock.json` but you're right in saying that if the version is old and doesn't look at `pacakge-lock.json`, then this might cause problems. I'm not sure. I feel like I'd rather cross that bridge when we get there, but I'm open to changing this if you feel strongly against that approach.

Reply:
I agree with you. Deal with it if it's a problem, not before. :+1: